org/mozilla/jss/util/NativeProxy.java

```diff
@@ -173,6 +197,21 @@ public final void clear() {
      * releaseNativeResources() gets called.
      */
     static HashSet<NativeProxy> registry = new HashSet<NativeProxy>();
+    static int registryIndex;
```

Comment:
We probably can use `AtomicInteger`:
```
static AtomicInteger counter = new AtomicInteger();
int index;

protected NativeProxy(byte[] pointer, boolean track) {
    index = counter.getAndIncrement();
}
public int hashCode() {
    return index;
}
```

Reply:
Yeah, `AtomicInteger` is probably the best bet. The [old `NativeProxy`](https://github.com/dogtagpki/jss/blob/v4.5.0/org/mozilla/jss/util/NativeProxy.java#L115-L137) used a random integer. :unicorn: